software/displayerang.livecodescript

```diff
@@ -0,0 +1,1402 @@
+      end if
+      put tL,tT,tR,tB into tNow["tooltipRect"]
+      set the rect of field "tooltipField" stack "HowstrBeta" to tNow["tooltipRect"]
+      --      set the topleft of field "tooltipField" stack "HowstrBeta" to tNow["moving"]
+      set the layer of field "tooltipField" stack "HowstrBeta" to top
+      set the visible of field "tooltipField" stack "HowstrBeta" to true
+      unlock screen
+   else
+      set the visible of field "tooltipField" stack "HowstrBeta" to false
+      put empty into tNow["tooltipRect"]
+   end if -- tooltip checked, and inside a graph element, and butt 1/3 not down
+   -------------------------------- end tooltip -----------------------------------
+   
+   -- (1c) if there's an anchor node, track new areas entered
+   if p1Down is empty then
+      -- either not inside a node, or mouse not down, or both
+      put empty into p1Enter
+   else if (pHist["inside"] is not empty) and (tNow["inside"] is empty) then
+      -- just entered background
+      put "background" into p1Enter
+   else if pHist["inside"] is not tNow["inside"] then
+      -- just entered a graph element
+      put tNow["inside"] into p1Enter
+   end if
+   
+   -- (2) build preview request
+   -- defaults to "nothing"
+   put "nothing" into tPreview
+   if p1Down is among the keys of pGUI["nodes"] then -- butt down in a node
+      -- anchor and target
+      if p1Enter is among the keys of pGUI["nodes"] and p1Enter is not p1Down then -- entered another node
+         put p1Down & "," & p1Enter into tPreview -- preview appropriate links/nodes between anchor & target
+      else if p1Enter is p1Down then -- reentered node where butt down
+         --         put p1Down & "," & "info" into tPreview -- ! swap 1
+         put p1Down into tPreview -- preview focus on this node
+      end if
+      -- link can't be target
+      if p1Enter is among the keys of pGUI["links"] then put "nothing" into tPreview -- don't do anything if entered a link
+      
+      if p1Enter is empty then -- never left the node where butt down
+         if p1Down is among the keys of pGUI["nodes"] then -- butt down in node (already checked for this in first if ?!)
+            --            if p1Down is pFocus then
+            --               put "nothing" into tPreview
+            --            else
+            --               --               put p1Down into tPreview -- ! swap 2
+            --               put p1Down & "," & "info" into tPreview
+            --            end if
+            put p1Down & "," & "info" into tPreview -- show info preview
+         else
+            put "nothing" into tPreview -- down in something not a node, no info preview
+         end if
+      end if
+      if p1Enter is "background" and p1Down is among the keys of pGUI["nodes"] then
+         put "nothing" into tPreview
+         if item 1 of the mouseLoc < item 1 of the rect of pGUI["nodes"][p1Down]["graphic"] then put p1Down & ",left" into tPreview
```

Comment:
Nice, thanks. The dictionary isn't particularly useful for discovering commands that already exist but you don't know the name of.